valid.go

```diff
@@ -313,6 +322,9 @@ func (d *decoder) wellformedHead() (t cborType, ai byte, val uint64, err error)
 		}
 		val = binary.BigEndian.Uint64(d.data[d.off : d.off+8])
 		d.off += 8
+		if err := d.acceptableFloat(math.Float64frombits(val)); err != nil {
+			return 0, 0, 0, err
+		}
```

Comment:
Same as comment above.  We need to check CBOR type before converting `val` to float.